### PR TITLE
Addition of a tutorial about the 1D analysis with the HLI

### PR DESCRIPTION
**Pull request by bkhelifi**
**Description**
This pull request is associated to the issue #2504 . It adds a tutorial about the 1D analysis with the HLI and adapts accordingly the `User Guide` page. 

**Dear reviewer**
As a non-expert of the HLI, I hope that I put the minimal command needed for the 1D analysis using the HLI.
I have a doubt about the path of the configuration file... to be checked.
I have noticed that there is a warning message during the FluxPoint estimation. For each point, there is the message:
`Reoptimize = False ignored for iminuit backend`
Is not normal??

### CODE REVIEW

**Review by registerrier — COMMENTED** [review]
The configuration file seems to be missing.

We have to consider how we use and demonstrate HLI in the tutorials. At the moment it is spread and not systematic. 

Adding a third example in the starting section would possibly require adding the API-based spectral analysis tutorial as well. But a getting started section should stay limited in scope.

Conversely, we might expose several use cases with API-based or HLI approaches. A possible approach being by associating a config file.

**Comment by bkhelifi** [comment]
When a notebook runs, where is it located? in `gammapy`, or in the location of the tutorial (`gammapy/docs/tutorials/starting` for this nb)?
The configuration file is here: `gammapy/analysis/config/example-1d.yaml`. So which path should I put?

**Comment by bkhelifi** [comment]
Dear all,
I do not know which path to add to the config yaml file. the CI failed because of that....

**Comment by adonath** [comment]
@bkhelifi Maybe just include the YAML config as a string in the notebook instead?

**Comment by bkhelifi** [comment]
The CI passes now after the change from yaml file to yaml strings...

File: docs/tutorials/analysis/1D/spectral_analysis_hli.ipynb [hunk 1]
```diff
@@ -0,0 +1,1264 @@
+{
```
Comment (registerrier) [inline docs/tutorials/analysis/1D/spectral_analysis_hli.ipynb:1]: <p>MAybe remove the comment in the <code>observations </code> section</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gammapy/gammapy/pull/4044/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (registerrier) [inline docs/tutorials/analysis/1D/spectral_analysis_hli.ipynb:1]: <p>This is already coded in the <code>AnalysisConfig</code> yaml.  This is not needed here. Maybe simply show that one can access the content this way, but don't reset everything </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gammapy/gammapy/pull/4044/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (registerrier) [inline docs/tutorials/analysis/1D/spectral_analysis_hli.ipynb:1]: <p>same</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gammapy/gammapy/pull/4044/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

**Review by registerrier — COMMENTED** [review]
One general question is whether or not to re-set configuration properties within the notebook.

I would avoid doing so.  You can possibly create substrings of the configuration to explain how things are set, first with observations then datasets etc. 

I would show how to change content programmatically only once to avoid people re-coding changes where the point is that they should design the yaml file correctly.

File: docs/tutorials/analysis/1D/spectral_analysis_hli.ipynb (hunk 1 above)
Comment (AtreyeeS) [inline docs/tutorials/analysis/1D/spectral_analysis_hli.ipynb:1]: <p>In "Proposed approach", "<code>Analysis</code> class to orchestrate data reduction" - it is not just data reduction, the full analysis is done with the Analysis class here, no?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gammapy/gammapy/pull/4044/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (bkhelifi) [reply docs/tutorials/analysis/1D/spectral_analysis_hli.ipynb:1]: <p>agree</p>

Comment (bkhelifi) [reply docs/tutorials/analysis/1D/spectral_analysis_hli.ipynb:1]: <p>done</p>

Comment (bkhelifi) [reply docs/tutorials/analysis/1D/spectral_analysis_hli.ipynb:1]: <p>removed</p>

Comment (bkhelifi) [reply docs/tutorials/analysis/1D/spectral_analysis_hli.ipynb:1]: <p>removed</p>

File: docs/tutorials/analysis/1D/config-1d.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+observations:
```
[the remark this answers is not in the document]
Comment (adonath) [reply docs/tutorials/analysis/1D/config-1d.yaml:1]: Yes, please remove....

File: docs/tutorials/analysis/1D/spectral_analysis_hli.ipynb [hunk 3]
```diff
@@ -0,0 +1,1165 @@
...
```
Comment (registerrier) [inline docs/tutorials/analysis/1D/spectral_analysis_hli.ipynb:1]: <p>Why is the copy needed?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gammapy/gammapy/pull/4044/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='41'/>

Comment (bkhelifi) [reply docs/tutorials/analysis/1D/spectral_analysis_hli.ipynb:1]: <p>removed</p>

[the remark this answers is not in the document]
Comment (bkhelifi) [reply docs/tutorials/analysis/1D/spectral_analysis_hli.ipynb:1]: <p>+1</p>

**Comment by bkhelifi** [comment]
When computing the flux points, one has still warning messages: ``Reoptimize = False ignored for iminuit backend``, one per point.
Can we catch them? If yes, can I try to do it in this PR??

**Comment by bkhelifi** [comment]
The test ``gammapy/visualization/tests/test_utils.py::test_plot_theta2_distribution`` is failing only for Windows:

```
             if np.any(np.less(err, -err, out=res, where=(err == err))):
                # like err<0, but also works for timedelta and nan.
>               raise ValueError(
                    f"'{dep_axis}err' must not contain negative values")
E               ValueError: 'yerr' must not contain negative values

C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\matplotlib\axes\_axes.py:3587: ValueError

```
Maybe there is some rounding that creates this error.

Should we add generically some tests in ``visualization.utils.py:plot_theta_squared_table`` on ``xerr`` and ``yerr`` such that there are \ge 0?

**Review by adonath — APPROVED** [review]
Thanks a lot @bkhelifi, I have no further comments here! My proposal would be to merge this PR ASAP, continue with #4080 and then spent some more time with general clean up and improvement of tutorials...

**Pull request merged**